tests/test_pythonpackage_basic.py

```diff
@@ -0,0 +1,310 @@
+    # TEST 1 from external ref
+    with mock.patch("pythonforandroid.pythonpackage."
+                    "extract_metainfo_files_from_package",
+                    fake_metadata_extract):
+        assert get_package_name("TeStPackaGe") == "testpackage"
+
+    # TEST 2 from a local folder, for which we'll create a fake package:
+    temp_d = tempfile.mkdtemp(prefix="p4a-pythonpackage-test-tmp-")
+    try:
+        with open(os.path.join(temp_d, "setup.py"), "w") as f:
+            f.write(textwrap.dedent("""\
+                from setuptools import setup
+                setup(name="testpackage")
+                """
+            ))
+        pkg_name = get_package_name(temp_d)
+        assert pkg_name == "testpackage"
+    finally:
+        shutil.rmtree(temp_d)
+
+
+def test_get_dep_names_of_package():
+    # TEST 1 from external ref:
+    # Check that colorama is returned without the install condition when
+    # just getting the names (it has a "; ..." conditional originally):
+    dep_names = get_dep_names_of_package("python-for-android")
+    assert "colorama" in dep_names
+    assert "setuptools" not in dep_names
+    dep_names = get_dep_names_of_package("python-for-android",
+                                         include_build_requirements=True)
+    assert "setuptools" in dep_names
+
+    # TEST 2 from local folder:
+    assert "colorama" in get_dep_names_of_package(local_repo_folder())
+
+    # Now test that exact version pins are kept, but others aren't:
+    test_fake_package = tempfile.mkdtemp()
+    try:
+        with open(os.path.join(test_fake_package, "setup.py"), "w") as f:
+            f.write(textwrap.dedent("""\
+            from setuptools import setup
+
+            setup(name='fakeproject',
+                  description='fake for testing',
+                  install_requires=['buildozer==0.39',
+                                    'python-for-android>=0.5.1'],
+            )
+            """))
+        # See that we get the deps with the exact version pin kept but
+        # the other version restriction gone:
+        assert set(get_dep_names_of_package(
+            test_fake_package, recursive=False,
+            keep_version_pins=True, verbose=True
+        )) == {"buildozer==0.39", "python-for-android"}
+
+        # Make sure we also can get the fully cleaned up variant:
+        assert set(get_dep_names_of_package(
+            test_fake_package, recursive=False,
+            keep_version_pins=False, verbose=True
+        )) == {"buildozer", "python-for-android"}
+
+        # Test with build requirements included:
+        dep_names = get_dep_names_of_package(
+            test_fake_package, recursive=False,
+            keep_version_pins=False, verbose=True,
+            include_build_requirements=True
+            )
+        assert len(
+            {"buildozer", "python-for-android", "setuptools"}.intersection(
+                dep_names
+            )
+        ) == 3  # all three must be included
```

Comment:
then why not directly:
```python
assert {"buildozer", "python-for-android", "setuptools"} == dep_names
```

Reply:
Ok so this is difficult. The short answer is because dep_names is `{"buildozer", "python-for-android", "setuptools", "wheel"}`.

The long answer is, any outside entity (as which I pretend to be testing, I'm not saying that is reasonable but that was the idea) should approach this with "I use setuptools to build, so `setuptools` should be in there" (with `include_build_requirements=True`), and then there's of course the other direct deps `buildozer` and `python-for-android`. For items that **I care about** (as a user) that would be `{"buildozer", "python-for-android", "setuptools"}` which is what the test verifies.

Now that `wheel` is added is due to the way pep517 operates which essentially needs it for building the metadata - that knowledge doesn't fall from the sky though, that is done manually in `pythonforandroid/pythonpackage.py`, *but* it feels like knowledge I just felt shouldn't be something the outside user worries about or something I would want to need to keep in sync as such an internal detail, and therefore I didn't want to test it here.

So that is why this is a subset test and the full set is not hardcoded. I'm sure you could argue for just writing the full practical result but it makes more sense to me the way it is, due to the explanation given above